### PR TITLE
Form field lengths

### LINKED ISSUE #1024: Styleguide: Form field lengths
Define common form field lengths, apply where missing, and document different lengths in the styleguide.

### PR DESCRIPTION
**Pull request by mparke**
closes #1024 

Addresses default form field widths, and makes adjustments to heights and line-heights.

For differences in checkboxes, see: #1088 
For differences in selects, see #1130 

Includes:
- Default form field width + some additional width classes
- bumps form field height to `32px`
- removed `line-height` definitions where possible
- replaced `label-col` with bootstrap's `col-form-label` where possible
- updates label color
- introduced `row-label` for non-form labels
- fixed `line-height`/`padding` on buttons and combos where I broke them
- `font-sm` changed to `13px`, `font-xs` to `12px`
- updates styleguide forms section ( see screenshots )

there's a lot of small line changes where I've replaced:
```
<div class="label-col">
  <label>
</div>
```

with

```
<label class="col-form-label"></label>"
```

Screenshots:

Before:
screen shot 2017-02-03 at 5 42 34 pm

After:
screen shot 2017-02-03 at 5 42 27 pm

Before:
screen shot 2017-02-03 at 5 42 48 pm

After: 
screen shot 2017-02-03 at 5 42 44 pm

Before:
screen shot 2017-02-03 at 5 43 02 pm

After:
screen shot 2017-02-03 at 5 42 54 pm

### CODE REVIEW

File: scss/components/form/form.scss [hunk 1]
```diff
@@ -4,37 +4,54 @@ form {
   }
 }
 
-// TOOD: remove input-container
```
Comment (mparke) [inline scss/components/form/form.scss]: default label and field width/height defined in here

File: scss/defaults.scss [hunk 2]
```diff
@@ -7,10 +7,10 @@ html {
   @include font-size(1.5);
```
Comment (mparke) [inline scss/defaults.scss]: font changes

File: scss/variables.scss [hunk 3]
```diff
@@ -43,8 +43,8 @@ $sidebar-width: 220px;
 
 /* BEGIN generic definitions */
 
-$input-height: 30px;
```
Comment (mparke) [inline scss/variables.scss]: default height and padding

File: scss/components/form/form.scss [hunk 4]
```diff
@@ -4,37 +4,54 @@ form {
...
-    &.input-group {
-        .btn {
-            position: relative;
-            left: -1px;
-        }
-    }
+.col-form-label {
+  color: $light-black; // $lightest-black
+  text-align: right;
+  line-height: 1.38;
```
Comment (mparke) [inline scss/components/form/form.scss]: I'm going to try a length unit here, there's a slight offset with field border last time I checked.

File: scss/components/Header.scss [hunk 5]
```diff
@@ -60,17 +63,18 @@ $header-padding: $grid-gutter-width-base / 2;
     }
 
     &-search {
-        $margin-tb: ($main-nav-height - $input-height) / 2;
-        height: $input-height;
-        margin: 0 $header-padding 0 0;
+        padding: 14px 10px;
+
+        input.form-control {
```
Comment (eatonphil) [inline scss/components/Header.scss]: Isn't this a duplicate of line 74?

File: scss/components/PrimaryPage.scss [hunk 6]
```diff
@@ -12,4 +12,43 @@
...
+
+    .input-group,
+    .btn-group,
+    .btn {
+        height: $form-field-height;
+    }
+
+    .btn {
+        border-radius: 0;
+    }
```
Comment (eatonphil) [inline scss/components/PrimaryPage.scss]: Why only define these here and not globally?

File: scss/components/PrimaryPage.scss [hunk 7]
```diff
@@ -12,4 +12,43 @@
...
+                background-color: $white;
+                width: 200px;
+            }
+        }
+
+        .input-group-btn {
+            height: $form-field-height;
+            float: left;
+        }
+    }
```
Comment (eatonphil) [inline scss/components/PrimaryPage.scss]: Ok sorry, why not define all these globally? I don't think thy should be here.

File: scss/components/PrimaryTable.scss [hunk 8]
```diff
@@ -36,4 +36,21 @@
...
+        margin-right: 15px;
+        -webkit-filter: contrast(70%) saturate(90%);
+        filter: contrast(70%) saturate(90%);
+        opacity: 0.7;
+        vertical-align: bottom;
+    }
+
+    .distro-style img {
+        margin-right: 15px;
+    }
```
Comment (eatonphil) [inline scss/components/PrimaryTable.scss]: :( Really would rather these specific rules not be in this PrimaryTable file.

File: scss/components/form/CheckboxInputCombo.scss [hunk 9]
```diff
@@ -1,16 +1,10 @@
...
 
     .Input {
+        display: inline-block;
         margin-left: 5px;
     }
-
-    .form-control {
-        display: inline;
-        width: auto;
-    }
```
Comment (eatonphil) [inline scss/components/form/CheckboxInputCombo.scss]: What existing file uses CheckboxInputCombo and does dropping these display rules affect how errors are displayed?

File: scss/components/form/RadioInputCombo.scss [hunk 10]
```diff
@@ -7,15 +7,9 @@
...
     }
 
     .Input {
         margin-left: 5px;
     }
-
-    .form-control {
-        display: inline;
-        width: auto;
-    }
```
Comment (eatonphil) [inline scss/components/form/RadioInputCombo.scss]: Same question here as the CheckboxInputCombo and error messages

File: scss/components/form/RadioSelectCombo.scss [hunk 11]
```diff
@@ -7,15 +7,9 @@
     .Radio,
     .Select {
         display: inline-block;
-        vertical-align: top;
```
Comment (eatonphil) [inline scss/components/form/RadioSelectCombo.scss]: Yeah?

File: scss/components/form/form.scss [hunk 12]
```diff
@@ -4,37 +4,53 @@ form {
...
+  width: 225px;
+}
+
+input.form-control.input-md {
+  width: 275px;
+}
+
+input.form-control.input-lg {
+  width: 325px;
+}
```
Comment (eatonphil) [inline scss/components/form/form.scss]: Why restrict these to `input`s only?

File: scss/defaults.scss [hunk 13]
```diff
@@ -26,6 +26,9 @@ body {
   @extend .font-normal;
   background: $light-gray;
   color: $black;
+  font-weight: normal;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
```
Comment (eatonphil) [inline scss/defaults.scss]: Why is this only a font-medium rule?

File: scss/defaults.scss [hunk 14]
```diff
@@ -85,3 +88,7 @@ hr {
 small {
   @extend .font-small;
 }
+
+strong {
+  font-weight: 600; // toned down from 'bolder'
```
Comment (eatonphil) [inline scss/defaults.scss]: 500? pretty please?

File: scss/utilities.scss [hunk 15]
```diff
@@ -15,6 +15,9 @@
 
 // TODO: prettify
 .alert {
+  border-radius: 0;
+  height: 32px;  // for now, match input height
```
Comment (eatonphil) [inline scss/utilities.scss]: These can / are easily be multiline, can we set line-height instead?

File: src/components/form/Radio.js [hunk 16]
```diff
@@ -2,24 +2,24 @@ import React, { PropTypes } from 'react';
...
-          type="radio"
-          className="form-control Radio-input"
-          checked={props.checked}
-          onChange={props.onChange}
-          value={props.value}
-        />
-        {props.label ? <span className="Radio-label">{props.label}</span> : null}
-      </label>
-    </div>
+    <label className="col-form-label Radio">
```
Comment (eatonphil) [inline src/components/form/Radio.js]: Can .Radio and .Radio-label just extend these instead?

File: src/linodes/components/Linode.js [hunk 17]
```diff
@@ -207,10 +207,11 @@ function renderRow(props) {
       <td>
         {renderBackupStatus(linode)}
       </td>
-      <td>
+      <td className="PrimaryTable-status-cell">
```
Comment (eatonphil) [inline src/linodes/components/Linode.js]: I'd prefer a less-specific rule like just right-aligning the last column or calling this PrimaryTable-rowOptions or literally just text-xs-right this td here.

File: src/linodes/linode/layouts/DashboardPage.js [hunk 18]
```diff
@@ -246,12 +244,10 @@ export class DashboardPage extends Component {
...
-              <div className="col-sm-4 label-col">
-                <label className="form-label" htmlFor="lish-input">
-                  Text console
-                </label>
-              </div>
-              <div className="col-sm-8 content-col right">
+              <label className="col-sm-4 col-form-label" htmlFor="lish-input">
+                Text console
+              </label>
+              <div className="col-sm-8  right">
```
Comment (eatonphil) [inline src/linodes/linode/layouts/DashboardPage.js]: You can drop all these `right` classes too (and `left` if they exist).

**Comment by eatonphil** [comment]
Dropped all `.input-container`? Woot!

File: scss/defaults.scss (hunk 13 above)
Comment (mparke) [inline scss/defaults.scss]: not sure what you mean, this is defined on `body`, you may be looking at collapsed lines in github?

Comment (eatonphil) [inline scss/defaults.scss]: Oops. Sorry about that, you're right.

Comment (eatonphil) [inline scss/defaults.scss]: What's with grayscale?

File: scss/utilities.scss (hunk 15 above)
Comment (mparke) [inline scss/utilities.scss]: multiline with newlines? if not, i'd expect multiple errors to be separate strings, and subsequently multiple divs/spans/etc.

What would the goal of line height be in that case?

Comment (eatonphil) [inline scss/utilities.scss]: If a form has multiple errors related to the entire form, we don't show a new alert box for each one. Right now we show a list (`ul`) of all the errors.

File: scss/defaults.scss (hunk 13 above)
Comment (mparke) [inline scss/defaults.scss]: it's the equivalent font-smoothing defined in `FF`, as far as I understand. however, let me back these effects out into another branch, we should compare separately

File: scss/utilities.scss (hunk 15 above)
Comment (mparke) [inline scss/utilities.scss]: Gotcha, let me find that, remove the height, and push a sample into the styleguide index

File: scss/components/form/RadioSelectCombo.scss (hunk 11 above)
Comment (mparke) [inline scss/components/form/RadioSelectCombo.scss]: I think so. Let me double check the groups + group errors.

File: src/components/form/Radio.js (hunk 16 above)
Comment (mparke) [inline src/components/form/Radio.js]: i was on the fence between implicit and explicit declarations (scss @extend vs additional class) in cases like this. let me look again.

Have you started to favor @extend ?

File: scss/components/form/CheckboxInputCombo.scss (hunk 9 above)
Comment (eatonphil) [inline scss/components/form/CheckboxInputCombo.scss]: Actually it's not the form-control lines that are worrying so much as the CheckboxInputCombo display:block (now).

File: src/components/form/Radio.js (hunk 16 above)
Comment (eatonphil) [inline src/components/form/Radio.js]: I'm also on the fence here. I can't argue strongly for either way, so might as well just keep it how you've got it I guess.

File: scss/components/form/RadioSelectCombo.scss (hunk 11 above)
Comment (mparke) [inline scss/components/form/RadioSelectCombo.scss]: Okay, I did find an issue here. Addressed with this change: https://github.com/linode/manager/pull/1135/commits/a6f765e2219a06af4846fba5582e84452f2ea759

making labels next to checks/radios vertical align middle now looks correct

**Comment by mparke** [comment]
Updated combo right-label alignment and added examples to index:

screen shot 2017-02-06 at 3 27 31 pm

**Comment by mparke** [comment]
Added multiple validation errors examples to index:

screen shot 2017-02-06 at 3 27 20 pm

**Pull request merged**